### PR TITLE
Obs web

### PR DESCRIPTION
**Pull request by sethgerou-noaa**
first pass at dcs - probably lots will change - but it's working and enough to demo + discuss.

### CODE REVIEW

File: apps/obs-web/src/views/DebrieferDcs.vue [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+      },
+      {
+        field: 'level',
+        header: 'Level',
+        key: 'level',
+        width: '60',
+        isEditable: true,
+        type: 'toggle',
+        listType: 'template',
+        list: [
```
Comment (melinashak-noaa) [inline apps/obs-web/src/views/DebrieferDcs.vue:108]: You could use Object.values(TripLevel) here to get the values you defined in your enum. Same for the rest of the enums

File: apps/obs-web/src/views/DebrieferDcs.vue [hunk 2]
```diff
@@ -0,0 +1,310 @@
+<template>
+  <div>
+    <div class="row">
+      <div class="col">
+        <p style="font-weight: bold; font-size: 20px; letter-spacing: .5px;">DCS/AFI - {{ observerName }}</p>
+      </div>
+      <div class="col">
+        <q-select v-model="selectedYear" :options="yearOptions" outlined label="year" dense options-dense style="max-width: 300px"></q-select>
```
[the remark this answers is not in the document]
Comment (sethgerou-noaa) [reply apps/obs-web/src/views/DebrieferDcs.vue:8]: I could see the default display being driven by the evaluation period, but I thought they needed a way to look at the entire year's dcs rows.

File: apps/obs-web/src/views/DebrieferDcs.vue [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+
+    const getYearOptions = () => {
+      const firstYear = '2000';
+      const currentYear = moment().format('YYYY');
+      for (let i = parseInt(firstYear, 10); i <= parseInt(currentYear, 10); i++) {
+        yearOptions.value.unshift(i.toString());
+      }
+    };
+
+    const openNewDebriefingTab = () => {
```
[the remark this answers is not in the document]
Comment (sethgerou-noaa) [reply apps/obs-web/src/views/DebrieferDcs.vue:195]: nope.  removing.

File: apps/obs-web/src/views/PrimeTable.vue [hunk 4]
```diff
@@ -153,6 +157,54 @@
       :columns="popupColumns"
       :uniqueKey="popupUniqueKey"
     />
+
+    <q-dialog v-model="dcsDetailsDialog" persistent position="bottom">
```
[the remark this answers is not in the document]
Comment (sethgerou-noaa) [reply apps/obs-web/src/views/PrimeTable.vue:161]: ha.  dare to dream :)

File: apps/obs-web/src/views/PrimeTable.vue [hunk 5]
```diff
@@ -564,6 +622,54 @@ export default createComponent({
...
+        dcsRow.value = buildRow(isAfi);
+        dcsDetailsDialog.value = true;
+    };
+
+    const onRowContextMenu = (event: any) => {
+        selectedRow.value = event.data;
+        cm.value.show(event.originalEvent);
+    };
+
+    const submit = async () => {
```
[the remark this answers is not in the document]
Comment (sethgerou-noaa) [reply apps/obs-web/src/views/PrimeTable.vue:663]: yes.  On the todo list.  Ideally the dcs rows would refresh when the user tabs over to the dcs tab.  Have you written anything that handles refresh (or some other behavior) on Prime TabPanel change tab?

File: apps/obs-web/src/views/PrimeTable.vue [hunk 6]
```diff
@@ -564,6 +622,54 @@ export default createComponent({
       return cols;
     }
 
+    const selectedRow: any = ref(null);
+    const cm: any = ref(null);
+    const menuModel = [
```
[the remark this answers is not in the document]
Comment (sethgerou-noaa) [reply apps/obs-web/src/views/PrimeTable.vue:628]: implemented

File: apps/obs-web/src/views/DebrieferDcs.vue (hunk 1 above)
Comment (sethgerou-noaa) [reply apps/obs-web/src/views/DebrieferDcs.vue]: implemented

File: apps/obs-web/src/views/DebrieferDcs.vue [hunk 7]
```diff
@@ -0,0 +1,310 @@
...
+      </div>
+      <div class="col" style="text-align: right">
+        <q-btn round icon="refresh" size="sm" color="primary" style="position: relative; top: 4px; margin-right: 10px" @click="refresh"></q-btn>
+        <q-btn
+          size="md"
+          color="primary"
+          dense
+          style="position: relative; top: 3px"
+          @click="addDcsRow"
+          >add dcs row</q-btn
```
[the remark this answers is not in the document]
Comment (sethgerou-noaa) [reply apps/obs-web/src/views/DebrieferDcs.vue:21]: implemented - add row now always handled by the context menu, which loads the dialog

File: apps/obs-web/src/views/PrimeTable.vue (hunk 5 above)
[the remark this answers is not in the document]
Comment (sethgerou-noaa) [reply apps/obs-web/src/views/PrimeTable.vue:688]: creating a new ticket for this

**Pull request merged**